extensions/ql-vscode/src/pure/log-summary-parser.ts

```diff
@@ -0,0 +1,31 @@
+import { EvaluatorLogData } from '../run-queries';
+
+/**
+ * A pure method that parses a string of evaluator log summaries into
+ * an array of EvaluatorLogData objects. 
+ * 
+ */
+ export function parseVisualizerData(logSummary: string): EvaluatorLogData[] {
+    // Remove newline delimiters because summary is in .jsonl format.
+    const jsonSummaryObjects: string[] = logSummary.split('\n\n');
+    const visualizerData: EvaluatorLogData[] = [];
+  
+    for (const obj of jsonSummaryObjects) {
+      const jsonObj = JSON.parse(obj);
+  
+      // Only convert log items that have an RA and millis field
+      // REVIEW: Not sure this is exactly what we want. Are we now excluding too many items?
+      // REVIEW: Is there a way to make this less brittle? 
+      if (jsonObj.ra != undefined && jsonObj.millis != undefined) {
+        const newLogData: EvaluatorLogData = {
+          queryName: jsonObj.queryCausingWork,
+          predicateName: jsonObj.predicateName,
+          timeInMillis: jsonObj.millis,
+          resultSize: jsonObj.resultSize
+          // TODO: need to also parse RA, pipeline arrays into the object. 
```

Comment:
Thanks for this comment, I hadn't considered saving something like the position in file in bytes. During 🍐 with @aeisenberg we discussed that this would be a good idea but will follow-on to the PR (and the front end visualizer PR) with this change, so that we can view simple/small logs in the visualizer first. The change is still under a canary flag 🐦 so we don't expect the memory pressure to be too pressing at the moment.